### PR TITLE
#160: Add missing features to deploy pipeline

### PR DESCRIPTION
**Pull request by mkjmdski**
closes https://github.com/vegaprotocol/jenkins-shared-library/issues/160

### CODE REVIEW

File: vars/pipelineDeploy.groovy [hunk 1]
```diff
@@ -180,6 +207,55 @@ void call() {
...
+                    withCredentials([sshCredentials]) {
+                        script {
+                            netSsh = { command ->
+                                return 'ssh -t -i $PSSH_KEYFILE $PSSH_USER@n04.$NET_NAME.vega.xyz "sudo' + command + '"'
+                            }
+                            newestFile = sh (
+                                script: netSsh 'ls -t /home/vega/.local/state/vega/node/checkpoints/ | head -n 1',
+                                returnStdout: true
+                            ).trim()
+                            version = sh (
```
Comment (daniel1302) [inline vars/pipelineDeploy.groovy:238]: let's add trim here

File: vars/pipelineDeploy.groovy [hunk 2]
```diff
@@ -180,6 +207,55 @@ void call() {
...
+                                script: netSsh "/home/vega/current/vega version | awk '{print $3}'"
+                            )
+                            sh netSsh "cp  /home/vega/.local/state/vega/node/checkpoints/${newestFile} /tmp/${newestFile}; chown `whoami`:`whoami` /tmp/${newestFile}"
+                            sh 'scp -i $PSSH_KEYFILE $PSSH_USER@n04.$NET_NAME.vega.xyz:/tmp/${newestFile} .'
+                            sh "mkdir -p checkpoint-store/Testnet-1/${version}"
+                            sh "mv ${newestFile} checkpoint-store/Testnet-1/${version}/"
```
Comment (daniel1302) [inline vars/pipelineDeploy.groovy:244]: We are using the Fairground directory: https://github.com/vegaprotocol/checkpoint-store/tree/main/Fairground

File: vars/pipelineDeploy.groovy [hunk 3]
```diff
@@ -180,6 +207,55 @@ void call() {
...
+                            dir('checkpoint-store'){
+                                sh "vegatools checkpoint --file 'Testnet-1/${version}/${newestFile}' --out 'Testnet-1/${version}/${newestFile}.json'"
+                                sshagent(credentials: ['vega-ci-bot']) {
+                                    sh 'git config --global user.email "vega-ci-bot@vega.xyz"'
+                                    sh 'git config --global user.name "vega-ci-bot"'
+                                    sh "git add Testnet-1/${version}"
```
Comment (daniel1302) [inline vars/pipelineDeploy.groovy:250]: Fairground directory

File: vars/pipelineDeploy.groovy [hunk 4]
```diff
@@ -180,6 +207,55 @@ void call() {
...
+                                script: netSsh "/home/vega/current/vega version | awk '{print $3}'"
+                            )
+                            sh netSsh "cp  /home/vega/.local/state/vega/node/checkpoints/${newestFile} /tmp/${newestFile}; chown `whoami`:`whoami` /tmp/${newestFile}"
+                            sh 'scp -i $PSSH_KEYFILE $PSSH_USER@n04.$NET_NAME.vega.xyz:/tmp/${newestFile} .'
+                            sh "mkdir -p checkpoint-store/Testnet-1/${version}"
+                            sh "mv ${newestFile} checkpoint-store/Testnet-1/${version}/"
+                            dir('checkpoint-store'){
+                                sh "vegatools checkpoint --file 'Testnet-1/${version}/${newestFile}' --out 'Testnet-1/${version}/${newestFile}.json'"
+                                sshagent(credentials: ['vega-ci-bot']) {
+                                    sh 'git config --global user.email "vega-ci-bot@vega.xyz"'
```
Comment (daniel1302) [inline vars/pipelineDeploy.groovy:248]: Is jenkins able to checkout to the main branch or it will checkout to the specific commit?

**Review by daniel1302 — APPROVED** [review]
Let's make sure that other pipeline especially devnet is working fine before merge this

File: vars/pipelineDeploy.groovy (hunk 4 above)
Comment (mkjmdski) [reply vars/pipelineDeploy.groovy:248]: It goes to the main branch always -> https://github.com/vegaprotocol/jenkins-shared-library/pull/169/files#diff-91494a3a0a2f7511e3f572efd7eef6686f778cb908cad29732bcea055134ae10R127 and I think this is good behaviour to don't cause any conflicts

**Review by daniel1302 — APPROVED** [review]

**Pull request merged**